src/deskolem.ml

```diff
@@ -3,186 +3,216 @@ open! Lplib
+                (* check if [args] is already in [args_set]. *)
+                if List.exists (List.for_all2 (Eval.eq_modulo []) args) args_set
+                    (* or [t] is not f-frozen. *)
+                    || not (frozen vlist t) then
                     args_set
                 else
+                    (* add the arguments of [t] to the result. *)
                     args::args_set
             end
+        (* otherwise we build the new set of arguments the term [h]. *)
         else
             get_ui f vlist args_set h
-    | Meta _                    -> Common.Console.out 4 "[DEBUG] Meta NOT HANDLED: [%a]@." Print.pp_term t; assert false
-    | Patt _                    -> Common.Console.out 4 "[DEBUG] Patt NOT HANDLED: [%a]@." Print.pp_term t; assert false
-    | TEnv _                    -> Common.Console.out 4 "[DEBUG] TEnv NOT HANDLED: [%a]@." Print.pp_term t; assert false
-    | Wild                      -> Common.Console.out 4 "[DEBUG] Wild NOT HANDLED: [%a]@." Print.pp_term t; assert false
-    | LLet _                    -> Common.Console.out 4 "[DEBUG] LLet NOT HANDLED: [%a]@." Print.pp_term t; assert false
-    | TRef _                    -> Common.Console.out 4 "[DEBUG] TRef NOT HANDLED: [%a]@." Print.pp_term t; assert false (* is not handled in the encoding. *)
-
-(** [get_option opt] returns [x] if [opt] is of the shape `Some x`, raise an
-    Invalid_argument otherwise. *)
-let get_option = fun opt ->
-    match opt with
-    | Some(x)   -> x
-    | None      -> raise (Invalid_argument "The option is None.")
+    | Meta _        -> assert false
+    | Patt _        -> assert false
+    | TEnv _        -> assert false
+    | Wild          -> assert false
+    | LLet _        -> assert false
+    | TRef _        -> assert false (* is not handled in the encoding. *)
 
 (** [size t] return the size of the term [t]. *)
 let rec size : term -> int = fun t ->
-  List.fold_right (fun a l -> size a + l) (snd (Term.get_args t)) 1
+  List.fold_right (fun a l -> size a + l) (Term.get_args t |> snd) 1
 
 (** [size_args f args] return the size of the term [f a0 a1 ... an] where args
     = [a0; a1; ...; an]. *)
 let size_args : sym -> term list -> int = fun f args ->
-    size (Term.add_args (mk_Symb(f)) args)
+    size (Term.add_args (mk_Symb f) args)
 
 (** [subst_var x m n] substitutes the variable [x] with the term [n] inside the
     term [m].*)
 let subst_var : term Bindlib.var -> term -> term -> term = fun x m n ->
-    Common.Console.out 4 "subst_var [ %a / %a] %a@." Print.pp_term n Print.pp_var x Print.pp_term m;
     Bindlib.subst (Bindlib.unbox (Bindlib.bind_var x (lift m))) n
 
-(** [subst_mvar x m n] substitutes all variables that [x] contains with the terms
-    of [n] inside the term [m]. *)
+(** [subst_mvar x m n] substitutes all variables that [x] contains with the 
+    terms of [n] inside the term [m]. *)
```

Comment:
returns the term [t] with the variables of [xs] replaced by [ts].
Note that [xs] and [ts] should have the same length.